app/assets/javascripts/toc.js

```diff
@@ -1,28 +1,88 @@
-$(document).ready(function() {
-  let element = $('#toc')
-  if(element.length > 0) {
-    let selectedId = element.data('selected')
-    let baseUrl = element.data('url')
-    let initialUrl = `${baseUrl}?node=${selectedId}&full=true`
-
-    $.getJSON(
-      initialUrl,
-      function(data) {
-        element.tree({
-          data: data
-        });
-      }).done(function() {
-        let selectedNode = element.tree('getNodeById', selectedId);
-        element.tree('selectNode', selectedNode);
-    });
+class TocBuilder {
```

Comment:
Suggestion: Should we change the extension of this file to `.es6`? Also, should this be in `app/javascript`? I don't know if we've gotten webpack running with this project yet, though. If not we don't have to worry about it.

Reply:
Using the `.es6` extension doesn't work. Compilation throws an error about babelHelpers that I can't resolve.

I can try webpack and see how that works.